From 5358fa5f2ce0b796ce7fcd241dbe58e0a8be7009 Mon Sep 17 00:00:00 2001
From: Tang Yizhou <tangyizhou@huawei.com>
Date: Thu, 30 Dec 2021 16:27:16 +0800
Subject: [PATCH] share_pool: Show sp vmflags in /proc/$pid/smaps
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 5358fa5f2ce0b796ce7fcd241dbe58e0a8be7009


ascend inclusion
category: Feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I4NDAW
CVE: NA

-------------------

e80000600000-e80000603000 rw-s 00600000 00:05 1025
/sp_group_1 (deleted)
	Size:                 12 kB
	KernelPageSize:        4 kB
	MMUPageSize:           4 kB
	Rss:                   0 kB
	Pss:                   0 kB
	Shared_Clean:          0 kB
	Shared_Dirty:          0 kB
	Private_Clean:         0 kB
	Private_Dirty:         0 kB
	Referenced:            0 kB
	Anonymous:             0 kB
	LazyFree:              0 kB
	AnonHugePages:         0 kB
	ShmemPmdMapped:        0 kB
	Shared_Hugetlb:        0 kB
	Private_Hugetlb:       0 kB
	Swap:                  0 kB
	SwapPss:               0 kB
	Locked:                0 kB
	THPeligible:    0
	VmFlags: rd wr sh mr mw me ms pf io dc de nr dd sp
                                                        ~~

Signed-off-by: Tang Yizhou <tangyizhou@huawei.com>
Signed-off-by: Wang Wensheng <wangwensheng4@huawei.com>
Reviewed-by: Kefeng Wang<wangkefeng.wang@huawei.com>
Reviewed-by: Weilong Chen <chenweilong@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 fs/proc/task_mmu.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/fs/proc/task_mmu.c b/fs/proc/task_mmu.c
index 464562ad470f..4da09b397370 100644
--- a/fs/proc/task_mmu.c
+++ b/fs/proc/task_mmu.c
@@ -663,6 +663,9 @@ static void show_smap_vma_flags(struct seq_file *m, struct vm_area_struct *vma)
 #endif /* CONFIG_ARCH_HAS_PKEYS */
 #ifdef CONFIG_USERSWAP
 		[ilog2(VM_USWAP)]	= "us",
+#endif
+#ifdef CONFIG_ASCEND_SHARE_POOL
+		[ilog2(VM_SHARE_POOL)]	= "sp",
 #endif
 	};
 	size_t i;
-- 
2.26.2

